src/modules/python/pbs_hooks/PBS_alps_inventory_check.PY

```diff
@@ -155,11 +158,12 @@ def get_apstat_nids(msg):
     # not ship the subprocess module
     apstat_out = os.popen(cmd_apstat)
 
+    pattern = re.compile("(.*?)\s+(.*?)\s+(.*?)\s+(.*?)\s+(.*)")
+

```

Comment:
Yes, here '?' character makes it un-greedy match as it is coming after repetition qualifier '\*'. If it comes after '.' or any other character meaning would be different. I agree to use .+ to say 1 or more character instead of .*
 I tried with the named group, it is adding extra white spaces to the values that makes our next if statement to fail. I found this through the debugger 
-> status = apstat_record.group('state')
(Pdb) p status
'UP  '
-> mode = apstat_record.group('hw')
(Pdb) p mode
'B '
-> mode = apstat_record.group('hw')
(Pdb) n
-> if (status == "UP") and (mode == "B"):
(Pdb) p mode
'B '
It should work after trimming white spaces, but that will add a bit more complexity to the code.
I would prefer to keep the existing changes for groups.